source/adios2/toolkit/sst/dp/rdma_dp.c

```diff
@@ -916,8 +1348,15 @@ static DP_WSR_Stream RdmaInitWriterPerReader(CP_Services Svcs,
 
     for (i = 0; i < readerCohortSize; i++)
     {
-        fi_av_insert(Fabric->av, providedReaderInfo[i]->Address, 1,
-                     &WSR_Stream->ReaderAddr[i], 0, NULL);
+        if (fi_av_insert(Fabric->av, providedReaderInfo[i]->Address, 1,
+                         &WSR_Stream->ReaderAddr[i], 0, NULL) < 1)
+        {
+
+            Svcs->verbose(WS_Stream->CP_Stream, DPCriticalVerbose,
+                          "[RdmaInitWRiterPerReader] Failed inserting address "
+                          "into vector\n");
+            return NULL;
+        }
```

Comment:
While trying to figure out where things were going south, I added some additional checks to the libfabric dataplane. Not strictly related to this PR, but I think that it's still worthwhile to keep them now that they're written anyway?